cmd/launcher/main.go

```diff
@@ -62,22 +69,38 @@ func main() {
 	// fork-bombing itself. This is an ENV, because there's no
 	// good way to pass it through the flags.
 	if !env.Bool("LAUNCHER_SKIP_UPDATES", false) && !inBuildDir {
-		runNewerLauncherIfAvailable(ctx, systemSlogger.Logger)
+		if err := runNewerLauncherIfAvailable(ctx, systemSlogger.Logger); err != nil {
+			systemSlogger.Log(ctx, slog.LevelInfo,
+				"could not run newer version of launcher",
+				"err", err,
+			)
+			return 1
+		}
 	}
 
 	// if the launcher is being ran with a positional argument,
 	// handle that argument. Fall-back to running launcher
 	if len(os.Args) > 1 && !strings.HasPrefix(os.Args[1], `-`) {
 		if err := runSubcommands(systemSlogger); err != nil {
-			logutil.Fatal(logger, "err", fmt.Errorf("run with positional args: %w", err))
+			systemSlogger.Log(ctx, slog.LevelError,
+				"running with positional args",
+				"err", err,
+			)
+			return 1
 		}
-		os.Exit(0)
+		return 0
 	}
 
 	opts, err := launcher.ParseOptions("", os.Args[1:])
 	if err != nil {
-		level.Info(logger).Log("err", err)
-		os.Exit(1)
+		if launcher.IsInfoCmd(err) {
+			return 0
+		}
```

Comment:
I'd take it, but would be be cleaner if we handled the InfoCmd in the main loop, instead of `launcher.ParseOptions`?

Reply:
I'm not 100% sure I understand the suggestion.

This handles `launcher --version` (or `launcher --dev_help`), and then above (on line ~84ish, in the call to `runSubcommands`) we handle `launcher version` separately. I don't know if there's a clean way to combine those two cases?

Reply:
I'm not deep in the code, so this could be offbase. But maybe either:
1. `--dev_help` and `--version` become boolean flags, get returned and then we detect them and do the thing and exit?
2. they are not flags, and parsed as args?

Reply:
I think 1 isn't really different what than what we currently do -- just a slightly different set of return args for the `launcher.ParseOptions` function. `--dev_help` and `--version` are already boolean flags, and if they're set then we return the special `InfoCmd` error and handle it here.

My sense is that the implementation for 2 would feel a little unexpected, but maybe no more so than the current implementation is. The difficulty is mostly "we want to treat these two special flags as subcommands", and it's probably weird either way, regardless of whether we do it in the subcommand parsing section or we do it in the flag parsing section.

Reply:
Oh. I see why this is unclear. In (1) I mean returning it as a bool (or enum) in launcher.Options, instead of handling the the run there. But that's a bit of a deeper refactor, though possible a cleaner outcome. 🤷 I'd take this

Reply:
Do you think that refactor might be a good first issue? I think it'd be fairly simple to test appropriately, and the refactor would be limited in scope. I could write something up and file it if so.

Reply:
That's a good call. Yes.